scipy/optimize/_linprog.py

```diff
@@ -565,10 +566,12 @@ def linprog(c, A_ub=None, b_ub=None, A_eq=None, b_eq=None,
 
         sol = _linprog_highs(lp, solver=highs_solvers[meth],
                              **solver_options)
+        sol['status'], sol['message'] = (
+            _check_result(sol['x'], sol['fun'], sol['status'], sol['slack'],
+                          sol['con'], lp.bounds, tol, sol['message']))
```

Comment:
`_check_result` is performed for all the other solvers but it was missing for HiGHS